src/test/java/com/github/mtakaki/credentialstorage/hibernate/SessionHoldersTest.java

```diff
@@ -0,0 +1,119 @@
+package com.github.mtakaki.credentialstorage.hibernate;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import org.hibernate.SessionFactory;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import io.dropwizard.db.ManagedDataSource;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SessionHoldersTest {
+    private SessionHolders holders;
+
+    @Mock
+    private SessionFactory sessionFactory;
+
+    @Mock
+    private ManagedDataSource dataSource;
+
+    @Before
+    public void setup() {
+        this.holders = new SessionHolders(this.sessionFactory, this.dataSource);
+    }
+
+    @Test
+    public void testAdd() {
+        final UnitOfWorkAspect unitOfWork = mock(UnitOfWorkAspect.class);
+        this.holders.add(unitOfWork);
+    }
+
+    @Test
+    public void testCloseConnections() throws Exception {
+        this.holders.closeConnections();
+
+        verify(this.sessionFactory, times(1)).close();
+        verify(this.dataSource, times(1)).stop();
+    }
+
+    @Test
+    public void testRemoveWithLastUnitOfWorkAndSetToClose() throws Exception {
+        final UnitOfWorkAspect unitOfWork = mock(UnitOfWorkAspect.class);
+        this.holders.setCloseSession(true);
+
+        this.holders.remove(unitOfWork);
+
+        verify(this.sessionFactory, times(1)).close();
+        verify(this.dataSource, times(1)).stop();
+    }
+
+    @Test
+    public void testRemoveWithLastUnitOfWorkAndNotSetToClose() throws Exception {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://www.codacy.com/app/mitsuotakaki/CredentialStorageService-dw-hibernate/file/2455857661/issues/source?bid=3237716&fileBranchId=3250434#l60)